lib/stormpath-sdk/data_store.rb

```diff
@@ -305,10 +305,14 @@ def to_hash(resource)
           property = resource.get_property name, ignore_camelcasing: ignore_camelcasing
 
           # Special use cases are with Custom Data, Provider and ProviderData, their hashes should not be simplified
-          if property.kind_of?(Hash) and !resource_nested_submittable(resource, name)
+          if property.kind_of?(Hash) and !resource_nested_submittable(resource, name) and name != "items"
             property = to_simple_reference name, property
           end
 
+          if name == "items"
```

Comment:
Wouldn't this also apply to collection resources, that have an "items" property? Same for the condition above.

If this is the case, I don't think we would want to do this for every item in the collection.